lib/Backends/CPU/LLVMIRGen.cpp

```diff
@@ -723,9 +775,11 @@ void LLVMIRGen::generateLLVMIRForModule(llvm::IRBuilder<> &builder) {
     if (!isBundleCompatible) {
       emitDataParallelKernel(builder, bundle);
       bundle.clear();
+      createNewFnIfNecessary();
     }
     // Add a data parallel instruction to the bundle.
     bundle.push_back(I);
+    numInstrsInCurrentFn++;
```

Comment:
I would have expected the createNewFnIfNecessary function to be called each time we insert a function, but it seems that we carefully chose where we actually invoke that.
Could you add comments on why the calls to that function need to happen only at these specific locations?

Basically my worries is if we create some code in some other place, we may miss that we may have to split the function.

Reply:
We do not create multiple functions in LLVMIRGen currently. In We basically go over Glow IR instructions in generateLLVMIRForModule and emit LLVM IR for each of them into "main". The LLVM IR for a Glow IR instruction is typically a call of the implementation function from libjit. 

Sometimes, the "main" functions gets too big and in such cases this PR tries to split it.